### PR TITLE
Add new ax feature

### PR DESCRIPTION
**Pull request by FloraSauerbronn**
New ax feature for plot_transect function in plotting.py

### CODE REVIEW

**Comment by ocefpaf** [comment]
@FloraSauerbronn When creating multiple figures or, of the user already passed an inverted axis, the call to invert it will undo this. We need to check if the axis is inverted first, and then invert it if not. See https://github.com/pyoceans/python-ctd/blob/10883153b27d296ef038252fe8f8604c79cd3a49/ctd/plotting.py#L30-L33 for a pattern on how to achieve this.

**Comment by ocefpaf** [comment]
@FloraSauerbronn we also need to write some tests for this functionality. We can probably write two to cover the 3 cases in the notebook above.

- 1 is already covered
- 2 could be a figsize check to see if that is different from the default
- the last one, for the subplots, should probably the another image comparison check

File: gliderpy/plotting.py [hunk 1]
```diff
@@ -85,6 +85,9 @@ def plot_transect(
     cbar = fig.colorbar(cs, orientation="vertical", extend="both")
     cbar.ax.set_ylabel(var)
     ax.set_ylabel("pressure")
+
+    ax.set_ylim(ax.get_ylim()[0], 0)
```
Comment (ocefpaf) [inline gliderpy/plotting.py:89]: 👍🏽

File: tests/test_plotting.py [hunk 2]
```diff
@@ -1,36 +1,44 @@
...
-
     glider_grab.fetcher.dataset_id = "whoi_406-20160902T1700"
     df = glider_grab.to_pandas()
-    # Generate the plot
-    fig, ax = plot_track(df)
+    return df
 
+@pytest.mark.mpl_image_compare(baseline_dir=root.joinpath("baseline/"))
+def test_plot_track(glider_data):
+    # Generate the plot
```
Comment (ocefpaf) [inline tests/test_plotting.py:19]:
```suggestion
```

File: tests/test_plotting.py [hunk 3]
```diff
@@ -1,36 +1,44 @@
...
+    fig, ax = plot_track(glider_data)
     # Return the figure for pytest-mpl to compare
```
Comment (ocefpaf) [inline tests/test_plotting.py:21]:
```suggestion
```

File: tests/test_plotting.py [hunk 4]
```diff
@@ -1,36 +1,44 @@
...
-
-
 @pytest.mark.mpl_image_compare(baseline_dir=root.joinpath("baseline/"))
-def test_plot_transect():
-    glider_grab = GliderDataFetcher()
-
-    glider_grab.fetcher.dataset_id = "whoi_406-20160902T1700"
-    df = glider_grab.to_pandas()
+def test_plot_transect(glider_data):
     # Generate the plot
```
Comment (ocefpaf) [inline tests/test_plotting.py:26]:
```suggestion
```

File: tests/test_plotting.py [hunk 5]
```diff
@@ -1,36 +1,44 @@
...
-    fig, ax = plot_transect(df, 'temperature')
+    fig, ax = plot_transect(glider_data, 'temperature')
+    # Return the figure for pytest-mpl to compare
```
Comment (ocefpaf) [inline tests/test_plotting.py:28]:
```suggestion
```

File: tests/test_plotting.py [hunk 6]
```diff
@@ -1,36 +1,44 @@
...
+    return fig
 
+@pytest.mark.mpl_image_compare(baseline_dir=root.joinpath("baseline/"))
+def test_plot_transect_multiple_figures(glider_data):
+    # Generate the plot with multiple figures
```
Comment (ocefpaf) [inline tests/test_plotting.py:33]:
```suggestion
```

File: tests/test_plotting.py [hunk 7]
```diff
@@ -1,36 +1,44 @@
...
+    fig, (ax0, ax1) = plt.subplots(figsize=(15, 9), nrows=2, sharex=True, sharey=True)
+    glider_data.plot_transect(var="temperature", ax=ax0)
+    glider_data.plot_transect(var="salinity", ax=ax1, cmap=cmcrameri.cm.davos)
     # Return the figure for pytest-mpl to compare
```
Comment (ocefpaf) [inline tests/test_plotting.py:37]:
```suggestion
```

File: tests/test_plotting.py [hunk 8]
```diff
@@ -1,36 +1,44 @@
...
     return fig
+
+def test_plot_transect_size(glider_data):
+    # Generate the plot with a specific size
```
Comment (ocefpaf) [inline tests/test_plotting.py:41]:
```suggestion
```

**Review by ocefpaf — COMMENTED** [review]
@FloraSauerbronn this one looks good! Let's remove the comments that are describing what the code is doing and add some docstrings to these functions?

Docstrings can be simple descriptions of test.

**Comment by ocefpaf** [comment]
@FloraSauerbronn I believe that all the changes here are part of the merged #114, right?

**Comment by FloraSauerbronn** [comment]
> @FloraSauerbronn I believe that all the changes here are part of the merged #114, right?

Yes, all the chamges are in [`#114`](https://github.com/ioos/gliderpy/pull/114)

**Pull request closed without merging**